ownCloud/Viewer/WebViewDisplayViewController.swift

```diff
@@ -0,0 +1,112 @@
+//
+//  ImageDisplayViewController.swift
+//  ownCloud
+//
+//  Created by Pablo Carrascal on 30/08/2018.
+//  Copyright © 2018 ownCloud GmbH. All rights reserved.
+//
+
+import UIKit
+import ownCloudSDK
+import WebKit
+
+class WebViewDisplayViewController: DisplayViewController, DisplayExtension {
+	
+	static var displayExtensionIdentifier: String = "imageViewer"
+	static var supportedMimeTypes: [String] =
+		["image/jpeg",
+		 "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
+		 "application/vnd.openxmlformats-officedocument.wordprocessingml.document"]
```

Comment:
~~Please also add MIME Types for other supported formats. As far as I can tell, https://github.com/owncloud/core/blob/master/core/js/mimetypelist.js should be an exhaustive list of what to expect from the OC server.~~

Thinking about it, this isn't as future-proof as it could be.

So in the latest commit I added support for a `customMatcher` to OCExtension. That block gets the `OCExtensionContext` and the `OCExtensionPriority` computed by the default implementation. To add wild-card matching for f.ex. `image/*`, `video/*`, `audio/*` and `text/*`, all one would need to do is add a `customMatcher` block to the `OCExtension` instance representing WebViewDisplayViewController.

In that block, you'd need to check if the provided `OCExtensionPriority` is `OCExtensionPriorityNoMatch`. And if it is, check if the context's location.locationIdentifier begins with `image/`, `video/`, `audio/` or `text/`. If it is, the block could return `OCExtensionPriorityLocationMatch`, otherwise return `OCExtensionPriorityNoMatch`.

That block could be provided to `DisplayExtension.displayExtension` by adding another property to return a `customMatcher`.

That said, adding those MIME Types from https://github.com/owncloud/core/blob/master/core/js/mimetypelist.js that do not match one of the "wildcards" should still be done, especially for Office formats like PowerPoint.